### PR TITLE
cml02a_gl listing

### PR DESCRIPTION
**Pull request by BFalquet**
add listing and tests

thank you for the review.

### CODE REVIEW

File: R/cml02a_gl.R [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+#' @describeIn cml02a_gl Preprocessing
+#'
+#' @inheritParams cml02a_gl_main
+#' @inheritParams gen_args
+#' @returns the preprocessing function returns a `list` of `data.frame`.
+#'
+#' @export
+#'
+cml02a_gl_pre <- function(adam_db,
```
Comment (clarkliming) [inline R/cml02a_gl.R:24]: why pre functions and main functions have different default values (when using `run` they should have the same arguments)

File: R/cml02a_gl.R [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# cml02a_gl_main ----
+
+#' @describeIn cml02a_gl Main TLG function
+#'
+#' @inheritParams gen_args
+#' @returns the main function returns an `rlistings` or a `list` object.
+#' @export
+#'
+cml02a_gl_main <- modify_default_args(std_listing,
```
Comment (clarkliming) [inline R/cml02a_gl.R:9]: while reading `std_listing` in https://github.com/insightsengineering/chevron/blob/745303078cdc43d89886d9df7dd8980e20b371b0/R/utils.R#L66 I found that the `unique_rows` are set to `TRUE` by default. Please check if other listings should have this unique option

Comment (BFalquet) [reply R/cml02a_gl.R:9]: so far, only `ael01_nollt` and `cml02a_gl` require `unique_rows = TRUE`. I have introduced this argument in the standard listing and set it to `FALSE` by default.

**Comment by barnett11** [comment]
Hi @BFalquet ,
When running listing with `head` or `tail` it works fine, but when running in its entirety I get below error?
`Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 'toString': values must be length 1,
 but FUN(X[[1]]) result is length 0`

**Comment by BFalquet** [comment]
> Hi @BFalquet , When running listing with `head` or `tail` it works fine, but when running in its entirety I get below error? `Error in h(simpleError(msg, call)) : error in evaluating the argument 'x' in selecting a method for function 'toString': values must be length 1, but FUN(X[[1]]) result is length 0`

Sounds like an rlistings issue. Could you try to update `formatters` and `rlistings` (`remotes::github_install("insightsengineering/rlistings")`)

**Comment by BFalquet** [comment]
@Melkiades any idea ?

**Review by barnett11 — COMMENTED** [review]
HI @BFalquet ,
This looks perfect aside of the rlisting bug, the only suggestion I'd make is to not wrap the header as below, as this variable will always be the longest text and the header length is not a limiter here thanks,
image

**Comment by BFalquet** [comment]
> to not wrap the header as below, as this variable will always be the longest text and the header length is not a limiter here thanks,

Alright, line break has been removed

**Review by barnett11 — APPROVED** [review]
Looks great - thanks @BFalquet

**Pull request merged**